channeldb/addr_source_test.go

```diff
@@ -0,0 +1,109 @@
+package channeldb
+
+import (
+	"net"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	addr1 = &net.TCPAddr{IP: (net.IP)([]byte{0x1}), Port: 1}
+	addr2 = &net.TCPAddr{IP: (net.IP)([]byte{0x2}), Port: 2}
+	addr3 = &net.TCPAddr{IP: (net.IP)([]byte{0x3}), Port: 3}
+)
+
+// TestMultiAddrSource tests that the multiAddrSource correctly merges and
+// deduplicates the results of a set of AddrSource implementations.
+func TestMultiAddrSource(t *testing.T) {
+	t.Parallel()
+
+	var pk1 = newTestPubKey(t)
+
+	t.Run("both sources have results", func(t *testing.T) {
+		t.Parallel()
+
+		var (
+			src1 = newMockAddrSource(t)
```

Comment:
would need to call `AssertExpectations(t)` to assert the mockers are called as expected.

Reply:
ah yes indeed - good catch! 🙏

Reply:
added